agent/command/util.go

```diff
@@ -61,19 +62,47 @@ func expandModulePrefix(conf *internal.TaskConfig, module, prefix string, logger
 	return modulePrefix
 }
 
-// getJoinedWithWorkDir joins the conf.WorkDir A with B like this:
+// getWorkingDirectory joins the conf.WorkDir A with B like this:
 //
 //	if B is relative, return A+B.
 //	if B is absolute, return B.
 //
 // We use this because B might be absolute.
-func getJoinedWithWorkDir(conf *internal.TaskConfig, path string) string {
+func getWorkingDirectory(conf *internal.TaskConfig, path string) string {
 	if filepath.IsAbs(path) {
 		return path
 	}
 	return filepath.Join(conf.WorkDir, path)
 }
 
+// getWorkingDirectoryLegacy is a legacy function to get the working directory
+// for a path, enforce that the path is always prefixed with the task working
+// directory, and check that the directory exists. This is a legacy function, so
+// should not be used anymore. Commands that need to get the working directory
+// should instead use getWorkingDirectory.
+func getWorkingDirectoryLegacy(tc *internal.TaskConfig, dir string) (string, error) {
```

Comment:
LGTM -- just to check, would we still want to use this function if we wanted to do some handling around directories (in which case we might not want to actually deprecate but call forDirectories or something), or is that unnecessary bc we'd get errors when we actually tried to use the path?

Reply:
It kind of depends on the command, but generally, if the directory doesn't already exist but it has to exist, both functions will error, this one will just error out a little earlier than the other one. The reason this one is slightly less preferable is because 1. many commands do stuff to the working directory that doesn't mandate that the directory must exist (e.g. s3.put with `optional: true`) and 2. if the directory needs to exist, the newer one will still have some error about the directory not existing, just further down the line when it needs the directory to exist.